### ISSUE TITLE
Questions about "aux2_loss"

### ISSUE THREAD

**Issue by iwa-shi** [LabShuHangGU/CCA#3]
Hi,

Thank you for sharing your work! I really enjoyed reading your paper, and congratulations on its acceptance at NeurIPS 2024!

I have a couple of questions regarding "aux2_loss", which I believe corresponds to L_Aux  in Eq(9) of the paper:

https://github.com/LabShuHangGU/CCA/blob/e4696bbbddb4816433d0701676792da8c28328aa/train.py#L55

1. If this is meant to calculate the bit rate,  shouldn’t it use `torch.log(output["aux_likelihoods"]["y_cca"])` as in "bpp_loss"?
For example:
```
out["aux2_loss"] = (torch.log(output["aux_likelihoods"]["y_cca"]).sum() + torch.log(output["aux_likelihoods"]["y_aux"]).sum()) / (-math.log(2) * num_pixels)
```

2. What is the purpose of including `EntropyBottleneck` in `AuxEntropyModel`? In the main branch, `EntropyBottleneck` is only applied to `z`.

I look forward to your response!

**Comment by MichTheEncoder** [comment]
Thank you for reading our paper. `EntropyBottleneck` is the fully factorized density model for roughly estimating the probability distribution of the input of the layer. Here we expect to utilize the `EntropyBottleneck` to get a rough estimation of the distribution of original `y`. By optimizing the cross entropy of the distribution of "roughly estimated" `y` and the distribution of "aux-estimated" `y_hat` (i.e., "aux2_loss"), the model is supposed to let the auxiliary model to estimate the same latent representation y_hat as the major entropy model like what we said in our paper. We just need the gradient back-propagated to impact the auxiliary entropy model, so the estimate of `y` does not need to be that accurate.

**Comment by iwa-shi** [comment]
I understand. Thank you for the clarification!
